app/src/main/java/org/greenstand/android/TreeTracker/models/StepCounter.kt

```diff
@@ -0,0 +1,63 @@
+package org.greenstand.android.TreeTracker.models
+
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import org.greenstand.android.TreeTracker.preferences.Preferences
+import timber.log.Timber
+
+class StepCounter(
+    private val sensorManager: SensorManager,
+    private val preferences: Preferences
+) {
+
+    private val stepCounter = sensorManager.getDefaultSensor(Sensor.TYPE_STEP_COUNTER)
+    private val stepCountEventListener = StepCountEventListener()
+    private var enabled = false
+
+    var absoluteStepCount: Int?
+        get() = preferences.getInt(User.ABS_STEP_COUNT)
+        private set(value) = preferences.edit().putInt(User.ABS_STEP_COUNT, value ?: 0).apply()
+
+    var absoluteStepCountOnTreeCapture: Int?
+        get() = preferences.getInt(User.ABS_STEP_COUNT_ON_TREE_CAPTURE)
+        set(value) = preferences
+            .edit().putInt(User.ABS_STEP_COUNT_ON_TREE_CAPTURE, value ?: 0).apply()
+
+    val viewLifecycleObserver: LifecycleObserver = object : LifecycleObserver {
+
+        @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+        fun unregisterListener() {
+            Timber.d("StepCounter: unregister listener")
+            sensorManager.unregisterListener(stepCountEventListener)

```

Comment:
We can call disable here instead

Reply:
Or we could make the class implement LifecycleObserver and just annotate the 'disable' method